articles/spring-social-auth/index.md

```diff
@@ -0,0 +1,241 @@
+title: Getting started with Spring Boot social authentication
+description: This tutorial will go over Spring Boot social authentication with both google and github. Spring Boot social auth makes it possible for users to authenticate into Spring Boot applications using their existing social accounts.
+author: odhiambo-paul
+date: 2021-03-18T00:00:00-13:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/spring-social-auth/hero.jpg
+    alt: Spring Boot social authentication example image
+---
+
+### Spring Boot Social auth
+
+Spring Boot social auth makes it possible for users to authenticate into Spring Boot applications using their existing social accounts. Using Spring Boot social authentication allows developers to major in business logic rather than majoring in developing a custom authentication system.
+
+### Prerequisites
+
+1. Java developer kit [JDK](https://www.oracle.com/java/technologies/javase-downloads.html) installed on your computer.
+2. Knowledge of [Kotlin](https://kotlinlang.org/docs/home.html) programming language.
+3. Knowledge of [Spring Boot](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/) framework.
+
+### Creating the application
+
+We are going to use [Spring initializr](https://start.spring.io/) to create our Spring Boot application.
+
+1. Navigate to [Spring initializr](https://start.spring.io/) in your browser.
+2. Select Kotlin in the languages section.
+3. Add `Spring Web`, `OAuth2 Client`, and `Spring Boot DevTools` dependencies.
+4. Leave other configurations as default and click on generate the project.
+5. Unzip the downloaded project and open it in your favorite IDE. I will be using [Intelij IDEA community](https://www.jetbrains.com/idea/download/#section=linux) which is available for free.
+6. Sync the project with maven to download and all the dependencies.
+
+### Adding webjar dependencies
+
+Since we are going to need `Jquery` to build the frontend for our application, we need to add `webjar` dependencies to make `Jquery` available in our Spring Boot project.
+
+In the `pom.xml` add the dependencies below to include `Jquery` in our project.
+
+```xml
+      <dependency>
+            <groupId>org.webjars</groupId>
+            <artifactId>jquery</artifactId>
+            <version>3.4.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.webjars</groupId>
+            <artifactId>bootstrap</artifactId>
+            <version>4.3.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.webjars</groupId>
+            <artifactId>webjars-locator-core</artifactId>
+        </dependency>
+```
+
+### Spring Security Configuration
+
+1. In the project's root package, create a package named `config`.
+2. Inside the `config` package created above, create a new kotlin file named `WebConfig` and add the code snippets below.
+
+```kotlin
+import org.springframework.context.annotation.Configuration
```

Comment:
I think you can leave out the imports and when there is a chance of importing the wrong class, you can highlight it in the instructions. But they are not such great hinderance.

Reply:
@LinusMuema point noted.